src/status_im/contexts/wallet/wallet_connect/events/sessions.cljs

```diff
@@ -18,18 +18,26 @@
 
 (rf/reg-event-fx
  :wallet-connect/disconnect-dapp
- (fn [{:keys [db]} [{:keys [topic on-success on-fail]}]]
+ (fn [{:keys [db]} [{:keys [topic name on-success on-fail]}]]
    (let [web3-wallet    (get db :wallet-connect/web3-wallet)
          network-status (:network/status db)]
      (log/info "Disconnecting dApp session" topic)
      (if (= network-status :online)
        {:fx [[:effects.wallet-connect/disconnect
               {:web3-wallet web3-wallet
                :topic       topic
-               :on-fail     on-fail
+               :on-fail     (fn []
+                              (rf/dispatch [:centralized-metrics/track-event "dapp-disconnected"
+                                            {:dapp_name name
+                                             :result    :fail}])
+                              (when on-fail
+                                (on-fail)))
                :on-success  (fn []
                               (log/info "Successfully disconnected dApp session" topic)
                               (rf/dispatch [:wallet-connect/delete-session topic])
+                              (rf/dispatch [:centralized-metrics/track-event "dapp-disconnected"
```

Comment:
can we use keywords instead of strings for the metric id? e.g. `:metric/dapp-disconnected`

Reply:
I believe we can, but we don't need :metric/ namespace before event, do we?

Reply:
I used `:metric` namespace as you suggested, it will be easier with suggestion list when you type the namespace